### PR TITLE
Enable setting default position of the simulated robot using ros2_control URDF tag.

### PR DESCRIPTION
**Pull request by destogl**
Adds support for initial robot's position through ros2_control URDF tag. 
This is also compatible with "FakeHardware" and can be used within the driver if needed.﻿

Related PRs are:
- ros-controls/ros2_control#593
- ros-controls/ros2_control#598

The example joint definition looks like (example from UR driver):

```
     <joint name="${prefix}shoulder_pan_joint">
        <command_interface name="position"/>
        <state_interface name="position">
          <param name="initial_value">${initial_positions['shoulder_pan_joint']}</param>  <!-- initial position for Simulators and FakeHardware -->
        </state_interface>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
      </joint>
```

This feature is already released in ros2_control, but not synced yet.

I am not sure about one thing: Do we need a separate example for this?

### CODE REVIEW

**Comment by bmagyar** [comment]
Sounds like a good idea and would save some headache with artificial homing, etc.

What do you think if all the defaults were collected under a separate tag within the gazebo component?

hand-wavy example

```
<system name="gazebosystem">
  <initial_state>
    <joint name="shoulder_pan_joint" position="1.0 "/>
    <joint name="shoulder_tilt_joint" position="0.5 "/>
    <joint name="gripper_joint" position="1.0" effort="0.0 "/>
  </initial_state>
</system>
```

**Comment by destogl** [comment]
I am more on the second approach. Because then we can reuse this in other cases like `FakeSystem`. Moreover, the information is available for hardware drivers if needed. (I could imagine it in some cases at least.)

File: gazebo_ros2_control_demos/launch/cart_example_position_initial_position.launch.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2021 Stogl Robotics Consulting UG (haftungsbeschränkt)
```
Comment (ahcorde) [inline gazebo_ros2_control_demos/launch/cart_example_position_initial_position.launch.py:1]: Do you mind to change the copyright to open robotics ? You can add a line indicating that you are the author

Change the year too

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: gazebo_ros2_control_demos/launch/cart_example_position_initial_position.launch.py (hunk 1 above)
Comment (destogl) [reply gazebo_ros2_control_demos/launch/cart_example_position_initial_position.launch.py:1]: Please check my [comment here](https://github.com/ignitionrobotics/ign_ros2_control/pull/27#issuecomment-1017040056). I actually prefer to remove this two additional files completelly.

@bmagyar do you have any thoughts on it?

Comment (ahcorde) [reply gazebo_ros2_control_demos/launch/cart_example_position_initial_position.launch.py:1]: I'm fine removing the new files

Comment (bmagyar) [reply gazebo_ros2_control_demos/launch/cart_example_position_initial_position.launch.py:1]: the less content the better

**Review by bmagyar — APPROVED** [review]

**Comment by destogl** [comment]
@ahcorde I removed the new files as discussed. Can you please do one more check? After [Friday's sync](https://discourse.ros.org/t/preparing-for-final-rolling-sync-on-ubuntu-focal-2022-01-25/23920/2?u=destogl) this should be buildable.

**Review by ahcorde — APPROVED** [review]

**Comment by ahcorde** [comment]
@destogl can you merge with master?

**Comment by destogl** [comment]
> @destogl can you merge with master?

I just rebased it. can you please enable workflows again.

**Comment by ahcorde** [comment]
Just one last thing @destogl , linters! https://github.com/ros-simulation/gazebo_ros2_control/runs/4966401765?check_suite_focus=true

**Comment by destogl** [comment]
@ahcorde done

**Pull request merged**